gradio/examples.py

```diff
@@ -296,6 +296,8 @@ async def cache(self) -> None:
                 if self.batch:
                     output = [value[0] for value in output]
                 cache_logger.flag(output)
+            Context.root_block.dependencies.remove(dependency)
```

Comment:
It might be good to add a comment here similar to what you added in `blocks.py` to explain why we're removing the dependency

Reply:
Done - thanks @abidlabs !